tensorboard/webapp/runs/views/runs_table/sorting_utils.ts

```diff
@@ -0,0 +1,131 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  SortingInfo,
+  SortingOrder,
+  TableData,
+} from '../../../widgets/data_table/types';
+import {ExperimentAlias} from '../../../experiments/types';
+
+enum UndefinedStrategy {
+  BEFORE,
+  AFTER,
+}
+
+interface SortOptions {
+  insertUndefined: UndefinedStrategy;
+}
+
+const POTENTIALLY_NUMERIC = new Set(['string', 'number']);
+
+const DEFAULT_SORT_OPTIONS: SortOptions = {
+  insertUndefined: UndefinedStrategy.AFTER,
+};
+
+export function parseNumericPrefix(value: string | number) {
+  if (typeof value === 'number') {
+    return isNaN(value) ? undefined : value;
+  }
+
+  if (!isNaN(parseInt(value))) {
+    return parseInt(value);
+  }
+
+  for (let i = 0; i < value.length; i++) {
+    if (isNaN(parseInt(value[i]))) {
+      if (i === 0) return;
+      return parseInt(value.slice(0, i));
+    }
+  }
+
+  return;
+}
+
+export function sortTableDataItems(
+  items: TableData[],
+  sort: SortingInfo
+): TableData[] {
+  const sortedItems = [...items];
+
+  sortedItems.sort((a, b) => {
+    let aValue = a[sort.name];
+    let bValue = b[sort.name];
+
+    if (sort.name === 'experimentAlias') {
+      aValue = (aValue as ExperimentAlias).aliasNumber;
+      bValue = (bValue as ExperimentAlias).aliasNumber;
+    }
+
+    if (aValue === bValue) {
+      return 0;
+    }
+
+    if (aValue === undefined || bValue === undefined) {
+      return orderFromLocalComparison(aValue, bValue);
+    }
+
+    if (
+      POTENTIALLY_NUMERIC.has(typeof aValue) &&
+      POTENTIALLY_NUMERIC.has(typeof bValue)
+    ) {
+      const aPrefix = parseNumericPrefix(aValue as string | number);
+      const bPrefix = parseNumericPrefix(bValue as string | number);
+      // Show runs with numbers before to runs without numbers
+      if (
+        (aPrefix === undefined || bPrefix === undefined) &&
+        aPrefix !== bPrefix
+      ) {
+        return orderFromLocalComparison(aPrefix, bPrefix, {
+          insertUndefined: UndefinedStrategy.BEFORE,
+        });
+      }
+      if (aPrefix !== undefined && bPrefix !== undefined) {
+        if (aPrefix === bPrefix) {
+          const aPostfix =
+            aValue.toString().slice(aPrefix.toString().length) || undefined;
+          const bPostfix =
+            bValue.toString().slice(bPrefix.toString().length) || undefined;
+          return orderFromLocalComparison(aPostfix, bPostfix, {
+            insertUndefined: UndefinedStrategy.BEFORE,
+          });
+        }
+
+        return orderFromLocalComparison(aPrefix, bPrefix);
+      }
+    }
+
+    return orderFromLocalComparison(aValue, bValue);
+  });
+  return sortedItems;
+
+  function orderFromLocalComparison(
```

Comment:
I do not understand the name of this function. Maybe just call it compareValues?

Reply:
Alright, `compareValues` it is.